crates/ruff/src/rules/pydocstyle/helpers.rs

```diff
@@ -62,3 +69,36 @@ pub fn normalize_word(first_word: &str) -> String {
         .replace(|c: char| !c.is_alphanumeric(), "")
         .to_lowercase()
 }
+
+/// Check decorator list to see if function should be ignored.
+pub fn should_ignore_definition(
+    checker: &Checker,
+    definition: &Definition,
+    ignore_decorators: &BTreeSet<String>,
+) -> bool {
+    if let DefinitionKind::Function(parent)
+    | DefinitionKind::NestedFunction(parent)
+    | DefinitionKind::Method(parent) = definition.kind
+    {
+        for decorator in cast::decorator_list(parent) {
+            let callable = map_callable(decorator);
+            if let Some(call_path) = compose_call_path(callable) {
```

Comment:
Why check twice? Can we not just use the second version (going through `resolve_call_path`)?

Reply:
That's not catching the `ignored_decorator` here:

https://github.com/charliermarsh/ruff/blob/cd8be8c0becbbcb5397d65b957c4b1821a600382/crates/ruff/resources/test/fixtures/pydocstyle/D.py#L409-L422

I guess because `BindingKind::FunctionDefinition` is mapped to `None` here:

https://github.com/charliermarsh/ruff/blob/d5abcda35ae58cedac2da2536bbecff7af40f824/crates/ruff/src/checkers/ast.rs#L248-L285

I can try fixing that if it makes sense.

Reply:
Ohh interesting. I'd actually vote _not_ to support the former, at least for now... It seems rare that you'd define a decorator that you'd want to ignore in this way within the same file that it's used, and it's consistent with the rest of these configuration options that they rely on import tracking. If we want to support inter-file definitions like this, we probably want to support them across-the-board, rather than for this one option. Does that make sense?

Reply:
Yeah, that makes sense. I guess users could still ignore their custom decorator in other modules where it's imported.